Mathlib/Data/Matrix/Rank.lean

```diff
@@ -274,4 +275,40 @@ theorem rank_eq_finrank_span_row [LinearOrderedField R] [Finite m] (A : Matrix m
   rw [← rank_transpose, rank_eq_finrank_span_cols, transpose_transpose]
 #align matrix.rank_eq_finrank_span_row Matrix.rank_eq_finrank_span_row
 
+section MulNonsingInv
+
+/-! ### Lemmas about pre or post multiplying by an invertible matrix. -/
+
+/-- Pre (Left) multiplying by an invertible matrix does not change the rank -/
+lemma rank_mul_IsUnit {m n R: Type}
+  -- [Fintype m] -- not needed according to linter
+  [Fintype n][DecidableEq n][CommRing R]
+  (A: Matrix n n R)(B: Matrix m n R)(hA: IsUnit A.det):
+  (B⬝A).rank = B.rank := by
+  rw [Matrix.rank, mulVecLin_mul, LinearMap.range_comp_of_range_eq_top, ←Matrix.rank ]
+  rw [LinearMap.range_eq_top]
```

Comment:
add to previous line

Reply:
I am depending on the `rfl` implicit in rw on line 288. I cannot merge the following rewrite because then there will be no `rfl` for the `rank B = rank B` goal.

Reply:
I can add a comment to clarify that if it helps?